ui/server/src/test/scala/pl/touk/nussknacker/ui/validation/ProcessValidationSpec.scala

```diff
@@ -248,6 +249,28 @@ class ProcessValidationSpec extends FunSuite with Matchers {
     validationResult.saveAllowed shouldBe true
   }
 
+  test("invalid disabled nodes don't invalidate process") {
```

Comment:
Can you provide more tests for scenarios with more complex connections like `Switch`, `Split` and `Joins` to prove that this recursion really works?